pennylane_lightning/src/Gates.cpp

```diff
@@ -220,6 +230,23 @@ Pennylane::RotationXGate::RotationXGate(double rotationAngle)
       js, c }
 {}
 
+const tuple<vector<CplxType>, double> Pennylane::RotationXGate::generator{
+    { 0, 1,
+      1, 0 }, -0.5
```

Comment:
I was initially worried about having the shift parameter of the generator hard coded here, and was thinking we could pass this from PennyLane to the backend as this would be more maintainable.

However, I doubt these numbers will change once fixed since it would change the results of any simulation that uses that gate. So maybe there is no maintainability worry! And hardcoding is better for speed!

So TLDR, looks good!

Reply:
Perhaps we should just make the generator and prefactor separate attributes?